lib/modules/versioning/debian/index.spec.ts

```diff
@@ -0,0 +1,368 @@
+    ${['8', '9', '10', '11']}                           | ${'foobar'}   | ${null}
+    ${['8', '9', '10', '11']}                           | ${'11'}       | ${'11'}
+    ${['8', '9', '10', '11']}                           | ${'10'}       | ${'10'}
+    ${['8', '9', '10', '11']}                           | ${'4'}        | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'2020.04'}  | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'foobar'}   | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'bullseye'} | ${'bullseye'}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'buster'}   | ${'buster'}
+    ${['jessie', 'stretch', 'buster', 'stable']}        | ${'stable'}   | ${'stable'}
+    ${['jessie', 'stretch', 'oldstable', 'bullseye']}   | ${'buster'}   | ${'oldstable'}
+    ${['jessie', 'oldoldstable', 'buster', 'bullseye']} | ${'warty'}    | ${null}
+  `(
+    'getSatisfyingVersion($versions, "$range") === "$expected"',
+    ({ versions, range, expected }) => {
+      expect(debian.getSatisfyingVersion(versions, range)).toBe(expected);
+    }
+  );
+
+  test.each`
+    versions                                            | range         | expected
+    ${['8', '9', '10', '11']}                           | ${'2020.04'}  | ${null}
+    ${['8', '9', '10', '11']}                           | ${'foobar'}   | ${null}
+    ${['8', '9', '10', '11']}                           | ${'11'}       | ${'11'}
+    ${['8', '9', '10', '11']}                           | ${'10'}       | ${'10'}
+    ${['8', '9', '10', '11']}                           | ${'4'}        | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'2020.04'}  | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'foobar'}   | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'bullseye'} | ${'bullseye'}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'buster'}   | ${'buster'}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'warty'}    | ${null}
+    ${['jessie', 'stretch', 'buster', 'stable']}        | ${'stable'}   | ${'stable'}
+    ${['jessie', 'stretch', 'oldstable', 'bullseye']}   | ${'buster'}   | ${'oldstable'}
+    ${['jessie', 'oldoldstable', 'buster', 'bullseye']} | ${'warty'}    | ${null}
+  `(
+    'minSatisfyingVersion($versions, "$range") === "$expected"',
+    ({ versions, range, expected }) => {
+      expect(debian.minSatisfyingVersion(versions, range)).toBe(expected);
+    }
+  );
+
+  test.each`
+    currentValue      | rangeStrategy | currentVersion | newVersion    | expected
+    ${undefined}      | ${undefined}  | ${undefined}   | ${'foobar'}   | ${'foobar'}
+    ${'stretch'}      | ${undefined}  | ${undefined}   | ${'11'}       | ${'bullseye'}
+    ${'stretch'}      | ${undefined}  | ${undefined}   | ${'bullseye'} | ${'bullseye'}
+    ${'stretch'}      | ${undefined}  | ${undefined}   | ${'stable'}   | ${'bullseye'}
```

Comment:
i don't think we should ever suggest updates to those stable tags. we only should update code name to code name, stable or code name to version (pin).

@rarkins WDYT?

Reply:
☝️ ❓